injectproxy/routes.go

```diff
@@ -405,3 +420,27 @@ func matchersToString(ms ...*labels.Matcher) string {
 	}
 	return fmt.Sprintf("{%v}", strings.Join(el, ","))
 }
+
+type QueryParseError struct {
+	msg string
+}
+
+func (e QueryParseError) Error() string {
+	return e.msg
+}
+
+func newQueryParseError(err error) QueryParseError {
+	return QueryParseError{msg: fmt.Sprintf("Error parsing query string %s", err.Error())}
+}
+
+type EnforceLabelError struct {
```

Comment:
Does this need to be exported or can we keep it lower case? If it really must be exported, then please add comment to document it
```suggestion
type enforceLabelError struct {
```

Reply:
Done